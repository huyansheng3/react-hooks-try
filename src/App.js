import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';
import Router from './router'
import { ThemeContext, LocaleContext } from './context'
import 'bulma/css/bulma.css';

function App() {
  console.log('app render')
  const [theme, changeTheme] = useState('red')
  const [locale, changeLocale] = useState('zh-CN')

  return (
    <div className="App">
      <header className="App-header">
        <ThemeContext.Provider value={{ theme, changeTheme }}>
          <LocaleContext.Provider value={{ locale, changeLocale }}>
            <Router></Router>
          </LocaleContext.Provider>
        </ThemeContext.Provider>
      </header>
    </div>
  );
}

export default App;
