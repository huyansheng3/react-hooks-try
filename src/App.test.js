import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('App renders snapshot', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});