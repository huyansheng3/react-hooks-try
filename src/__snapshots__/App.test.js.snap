// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="router"
    >
      <div>
        <h3>
          Begin
        </h3>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          prev
        </button>
        <button
          onClick={[Function]}
        >
          next
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`App renders snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="router"
    >
      <div>
        <h3>
          Begin
        </h3>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          prev
        </button>
        <button
          onClick={[Function]}
        >
          next
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`Tops renders snapshot 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "padding": "30px",
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
  <ul
    style={
      Object {
        "maxHeight": "300px",
        "overflowY": "scroll",
      }
    }
  />
</div>
`;
